### ISSUE TITLE
USER ISSUE: Solar Panel Works Underground

### ISSUE THREAD

**Issue by Trug42** [StrangeLoopGames/EcoIssues#4433]
**Version:** 0.7.1.0 beta

**Steps to Reproduce:**
1. Place a solar panel underground.
2. Enjoy the power of darkness

**Expected behavior:**
Not produced any energy due to being in a cave

**Actual behavior:**
Acted like it was under the open sky

**Comment by ecoemmeck** [comment]
Adding to this it probably should have a smaller power output at night

**Comment by StallEF** [comment]
@MazonDel  WindTurbine works Underground, maybe you can  fix it too

**Comment by SlayksWood** [comment]
- [ ] 1. Solar Generator:
Hard to test it in cave and undeground due to #21751, but i didn't undestand this one:
![image](https://user-images.githubusercontent.com/45708377/118236809-37914880-b49f-11eb-8b0f-4533ef46c7e1.png)
Footprint is more in 2x2 times than solar generator size. So if I build a room around solar generator it still works:
![image](https://user-images.githubusercontent.com/45708377/118237022-7921f380-b49f-11eb-92e9-e5dca812a444.png)

- [ ] 2. Wind Turbine:
Provide 100% power if you build a room without roof around it:
![image](https://user-images.githubusercontent.com/45708377/118238518-47aa2780-b4a1-11eb-8d80-d5ed888fca62.png)
![image](https://user-images.githubusercontent.com/45708377/118238532-4d077200-b4a1-11eb-8085-878397b81a71.png)
![image](https://user-images.githubusercontent.com/45708377/118238549-51338f80-b4a1-11eb-890c-119e66976adc.png)
![image](https://user-images.githubusercontent.com/45708377/118238563-55f84380-b4a1-11eb-9c65-924b1c61bb95.png)
It's a little weird to see the sky in demands when it needs wind from the side.

If you start to add roof it will be disabled at all:
![image](https://user-images.githubusercontent.com/45708377/118238738-9788ee80-b4a1-11eb-8cf6-c1a30a7b3fe8.png)
![image](https://user-images.githubusercontent.com/45708377/118238745-9e176600-b4a1-11eb-94ce-3509c9c4d4d9.png)

**Comment by MazonDel** [comment]
Yup, the wind power is intended to be part of the focus for Power Gen V2.

**Comment by Daniel-Magulado** [comment]
Tested on 0.9.4 Staging-2026.

Found a couple of issues with the latest implementation of Power Gen V2 PR.

### Disabled Wind Turbine due to single connecting blocks
For some reason, if you place 7 pieces of single-lane connecting blocks that are 12 blocks high in front of the wind turbine, the wind turbine gets disabled. See the image below.

![image](https://user-images.githubusercontent.com/77248866/122595154-9a779000-d09a-11eb-9a02-101076d0cdcb.png)

If you check the status of the Wind Turbine, it states:
![image](https://user-images.githubusercontent.com/77248866/122595319-dca0d180-d09a-11eb-9635-3524e7b78e70.png)
Even though the front of the Wind Turbine is clear as the sky.

### Max Output and Current Output
The readings for the Max Output and Current Output are always the same. Max Output should be the maximum potential that the Wind Turbine can produce while Current Output should be the current output it can produce.

![image](https://user-images.githubusercontent.com/77248866/122596339-44a3e780-d09c-11eb-95f5-578d4719a09e.png)

Furthermore, both readings are not being updated even though the status page is updated promptly.


https://user-images.githubusercontent.com/77248866/122596482-85036580-d09c-11eb-8ee6-b0dfb92a96aa.mp4

In the video above, the Maximum output should be 1,102.5 while the current output should be 1,014.3.
